openpype/hosts/nuke/api/workio.py

```diff
@@ -25,7 +25,13 @@ def open_file(filepath):
     # To remain in the same window, we have to clear the script and read
     # in the contents of the workfile.
     nuke.scriptClear()
-    nuke.scriptReadFile(filepath)
+    autosave = f"{filepath}.autosave"
+    autosave_prmpt = "Autosave detected.\nWould you like to load the autosave file?"       # noqa
+    if os.path.isfile(autosave) and nuke.ask(autosave_prmpt):
```

Comment:
I'm not sure if `open_file` should trigger UI message? Thinking about automations.

Reply:
it's weird but this UI implementation is to allow users to choose whether they want to use autosave script if there is one avaliable in the work directory. If it is removed and it will always load autosave script if there is one. 
maybe we can move this logic to `open_workfile` in `pipeline.py`.

Reply:
By the way, another option _could_ be to allow `.autosave` extensions to be visible in the workfiles tool? (Less artist friendly I suppose?)

Reply:
Maybe add kwargs to the function.
```python
def open_file(filepath, allow_autosave=True):
    ...
    if allow_autosave:
        autosave = f"{filepath}.autosave"
        autosave_prmpt = (
            "Autosave detected."
            "\nWould you like to load the autosave file?"
        )
        if os.path.isfile(autosave) and nuke.ask(autosave_prmpt):
            filepath = autosave
        nuke.scriptReadFile(filepath)
    ...
```
So it is at least possible in automations skip that option.

Reply:
> We could just have **kwargs in all the open_file interfaces but it'd just start becoming confusing across the different hosts since we don't know which we should pass to enforce no prompts. Also, if we were to add a toggle for allow autosave I'd actually assume that False or True would both avoid the prompt since they are explicit values either way.

The api defines that there must be 1 positional required argument that is filepath, any additional kwargs can be host specific, if we consider that the kwarg could be used only from the host itself. If not we could change it to `headless`. But that's discussion for different place.

Ad this PR, I couldn't find way how to find out, if nuke is running in batch mode (I didn't do deep search). If you know how, please do a suggestion to make it quicker.

Reply:
Couldn't find it quickly either - it seems in batch mode at least Nuke doesn't create a QApplication instance so you could check for the QApplication instance existing or not. But yeah, that's definitely not as clean as I'd expect.